pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -61,6 +63,41 @@ const (
 	OSDOverPVCLabelKey                  = "ceph.rook.io/pvc"
 )
 
+const (
+	activateOSDCode = `
+set -ex
+
+OSD_ID=%s
+OSD_UUID=%s
+OSD_STORE_FLAG="%s"
+TMP_DIR=$(mktemp -d)
+OSD_DATA_DIR=/var/lib/ceph/osd/ceph-"$OSD_ID"
+
+# create the osd directory
+mkdir -p "$OSD_DATA_DIR"
+
+# active the osd with ceph-volume
+ceph-volume lvm activate --no-systemd "$OSD_STORE_FLAG" "$OSD_ID" "$OSD_UUID"
+
+# copy the tmpfs directory to a temporary directory
```

Comment:
I'm confused why this is necessary, even with the description provided. Why copy from A to B, then B back to A? Where does the tempfs come from?

Reply:
The`tmpfs` gets mounted by c-v during the activation, then puts files in it. When the container is stopped, the tmpfs goes away and all the files with it, so the emptydir is empty when accessed from the main container.
To fix this, we need to copy the file (basically get ride of the tmpfs) so the files will persist once the activate container stops.

Does that help?

Reply:
Yes. That helps. Annoying, but I see why it's necessary.